controllers/badges.js

```diff
@@ -0,0 +1,123 @@
+const badgeQuery = require("../models/badges");
+const { fetchUser } = require("../models/users");
+const imageService = require("../services/imageService");
+
+/**
+ * Get badges data
+ * @param req {Object} - Express request object
+ * @param res {Object} - Express response object
+ * @returns {Array} - Return badges
+ */
+const getBadges = async (req, res) => {
+  try {
+    const allBadges = await badgeQuery.fetchBadges(req.query);
+    return res.json({
+      message: "Badges returned successfully!",
+      badges: allBadges,
+    });
+  } catch (error) {
+    logger.error(`Error while fetching all badges: ${error}`);
+    return res.boom.serverUnavailable("Something went wrong please contact admin");
+  }
+};
+
+const getUserBadgeIds = async (req, res) => {
+  try {
+    const { userExists, badgeIds } = await badgeQuery.fetchUserBadgeIds(req.params.username);
+    let responseMsg = "";
+    if (userExists) {
+      responseMsg = badgeIds.length !== 0 ? "User badges returned successfully!" : "This user does not have any badges";
+      return res.json({ message: responseMsg, badgeIds });
+    } else {
+      return res.boom.notFound("The user does not exist");
+    }
```

Comment:
we know that badgeIds is an array, so we can return it directly with the message badges returned successfully, and in the frontend we can check if the array is empty and change the message accordingly, also the following would be much cleaner

c
```suggestion
    const { userExists, badgeIds } = await badgeQuery.fetchUserBadgeIds(req.params.username);

    if (!userExists) {
       return res.boom.notFound("The user does not exist");
    }
      
      return res.json({ message: 'badges returned successfuly, badgeIds });
    
```

Reply:
I understood your point, and have make changes accordingly.